src/System.Management.Automation/engine/parser/Compiler.cs

```diff
@@ -4189,10 +4189,7 @@ private Expression GenerateDoLoop(LoopStatementAst loopStatement)
             // $foreach/$switch = GetEnumerator $enumerable
             var enumerable = NewTemp(typeof(object), "enumerable");
             temps.Add(enumerable);
-            if (generatingForeach)
-            {
-                exprs.Add(UpdatePosition(stmt.Condition));
-            }
+            exprs.Add(UpdatePosition(stmt.Condition));
```

Comment:
So simple 👍

Reply:
We generate code to update position in the script that is executing at various places, so when an error happens, the error can be associated with a position in the script. Here we didn't update the position for the `condition` when evaluating the condition from a switch statement.

Reply:
Interesting that the previous code went out of its way to only work for `foreach` loops

Reply:
Yes, I wonder why it was so? If we didn't have tests for that previously didn't we break something now?

Reply:
Yeah, I did break something 😄
When stepping over the script in debugging, now the debugger stops on `switch ($b)` twice. I will revert this PR.

Reply:
Don't add me to CI list :smile: